src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsregisterstonomisupdate/services/CourtRegisterUpdateService.kt

```diff
@@ -37,43 +37,72 @@ class CourtRegisterUpdateService(
     "OTH" to "OTHER"
   )
 
-  fun updateCourtDetails(court: CourtUpdate): MapDifference<String, Any>? {
+  fun updateCourtDetails(court: CourtUpdate): List<MapDifference<String, Any>> {
     log.info("About to update court $court")
-    return updateCourt(court)
-  }
+    val diffs: MutableList<MapDifference<String, Any>> = mutableListOf()
 
-  private fun updateCourt(court: CourtUpdate): MapDifference<String, Any>? {
     courtRegisterService.getCourtInfoFromRegister(court.courtId)?.run {
       log.info("Found court register data {}", this)
 
-      val updatedCourtDataFromRegister = convertToPrisonCourtData(this)
-      log.debug("Transformed register data to prison data format {}", updatedCourtDataFromRegister)
-
-      val currentCourtDataInPrisonSystem = prisonService.getCourtInformation(courtId)
-      log.debug("Found prison data version of court {}", currentCourtDataInPrisonSystem)
+      // check if court has multiple addresses with sub codes
+      buildCourts(this).forEach {
+        diffs.add(processCourt(it))
+      }
+    }
+    return diffs
+  }
 
-      val currentCourtDataToCompare =
-        if (currentCourtDataInPrisonSystem != null) translateToSync(currentCourtDataInPrisonSystem) else null
-      val newCourtData = mergeIds(updatedCourtDataFromRegister, currentCourtDataToCompare)
+  private fun buildCourts(courtDto: CourtDto): List<CourtDataToSync> {
 
-      val diffs = checkForDifferences(currentCourtDataToCompare, newCourtData)
-      if (!diffs.areEqual()) {
-        log.info("$courtId: APPLY CHANGES=$applyChanges - Updating Prison System with court data. Changes {}", diffs)
+    if (courtDto.buildings.size < 2) {
+      return listOf(convertToPrisonCourtData(courtDto, courtDto.buildings))
+    }
```

Comment:
so I guess even if a single building incorrectly has a subcode, we will ignore and just treat it as an address. Which is good.